### PR TITLE
Add grpc service registration helper that adds default middleware

### PR DESCRIPTION
**Pull request by ecordell**
This helps to write registration functions for grpc services that shouldn't be used without certain middleware (i.e. validation)

### CODE REVIEW

**Review by jakedt — APPROVED** [review]

File: middleware.go [hunk 1]
```diff
@@ -21,8 +26,70 @@ type AuthlessHealthServer struct {
...
+			},
+		}
+		svcDesc.Methods[i] = wrapped
+	}
+	return &svcDesc
+}
+
+// WrapStreams wraps all streaming endpoints with the given list of interceptors
+// it returns a copy of the ServiceDesc with the new wrapped methods
+func WrapStreaems(svcDesc grpc.ServiceDesc, interceptors ...grpc.StreamServerInterceptor) (wrapped *grpc.ServiceDesc) {
```
Comment (jzelinskie) [inline middleware.go:70]: WrapStreams

File: middleware.go [hunk 2]
```diff
@@ -21,8 +26,70 @@ type AuthlessHealthServer struct {
...
+func (s *AuthlessHealthServer) SetServicesHealthy(svcDesc ...*grpc.ServiceDesc) {
+	for _, d := range svcDesc {
+		s.SetServingStatus(
+			d.ServiceName,
+			healthpb.HealthCheckResponse_SERVING,
+		)
+	}
+}
+
+// DefaultUnaryMiddleware is a suggested set of middleware
```
Comment (jzelinskie) [inline middleware.go:45]: >DefaultUnaryMiddleware is a recommended set of middleware that should each gracefully no-op if the middleware is not applicable.

I added this latter half of the sentence to make it clear that you don't need to be using protoc-gen-validate to use this default middleware.

File: middleware.go [hunk 3]
```diff
@@ -21,8 +26,70 @@ type AuthlessHealthServer struct {
...
+var DefaultUnaryMiddleware = []grpc.UnaryServerInterceptor{grpcvalidate.UnaryServerInterceptor()}
+
+// WrapMethods wraps all non-streaming endpoints with the given list of interceptors
+// it returns a copy of the ServiceDesc with the new wrapped methods
```
Comment (jzelinskie) [inline middleware.go:49]: These are two different sentences: add the punctuation and capitalization.

File: middleware.go [hunk 4]
```diff
@@ -21,8 +26,70 @@ type AuthlessHealthServer struct {
...
+				return handler(srv, ctx, dec, grpcmw.ChainUnaryServer(interceptor, chain))
+			},
+		}
+		svcDesc.Methods[i] = wrapped
+	}
+	return &svcDesc
+}
+
+// WrapStreams wraps all streaming endpoints with the given list of interceptors
+// it returns a copy of the ServiceDesc with the new wrapped methods
```
Comment (jzelinskie) [inline middleware.go:69]: These are two different sentences: add the punctuation and capitalization.

File: middleware.go [hunk 5]
```diff
@@ -21,8 +26,70 @@ type AuthlessHealthServer struct {
...
+			Handler: func(srv interface{}, stream grpc.ServerStream) error {
+				return chain(srv, stream, info, handler)
+			},
+		}
+		svcDesc.Streams[i] = wrapped
+	}
+	return &svcDesc
+}
+
+// NoopUnaryInterceptor doesn't modify the request/response
```
Comment (jzelinskie) [inline middleware.go:92]: >... is a gRPC middleware that does not do anything.

**Review by jzelinskie — CHANGES_REQUESTED** [review]
You need to sign off on your commits and maybe we should add extra sentences to each doc comment just to give a hint of _when_ you'd want to use these functions.

File: middleware.go [hunk 6]
```diff
@@ -21,8 +26,70 @@ type AuthlessHealthServer struct {
...
+func (s *AuthlessHealthServer) SetServicesHealthy(svcDesc ...*grpc.ServiceDesc) {
+	for _, d := range svcDesc {
+		s.SetServingStatus(
+			d.ServiceName,
+			healthpb.HealthCheckResponse_SERVING,
+		)
+	}
+}
+
+// DefaultUnaryMiddleware is a recommended set of middleware that should each gracefully no-op if the middleware is not applicable.
```
Comment (jzelinskie) [inline middleware.go:45]: This doesn't look like 80char lines.
I need to fix CI to actually lint things correctly.
I can fix it in a follow up.

**Review by jzelinskie — APPROVED** [review]

File: middleware.go (hunk 6 above)
Comment (ecordell) [reply middleware.go]: fixed - also added a test

**Pull request closed without merging**